### PR TITLE
fixes isinf macro for windows hosts

### LINKED ISSUE #5812: isnum_safe() might not work right on Windows
One of the Elders™ dropped this important lore which I haven't had the chance to verify:
image

All of the compiler bots use Linux so isnum_safe() was written with the Linux behavior in mind. We need to check Windows behavior and fix the macro if necessary.

### PR DESCRIPTION
**Pull request by MNarath1**
## About The Pull Request

Original behavior for isinf did only account for the behavior of linux hosts and was unable to detect infinities on windows this PR fixes this by changing the check ~~and implementing additional checks for windows hosts~~ so it reliably works for windows and linux.
This is only a temporary fix as there seems to be already a native solution for the next major beyond version in work.
Closes https://github.com/BeeStation/BeeStation-Hornet/issues/5812

Thanks to kuberoot for helping to profile it:
~~while this does fix the problem turns out it also is slower than the original approach(that currently doesn't work on windows)
the difference is in around 1000000 loops the difference is 2.4 seconds for the new methode to 0.5 of the original methode so there is a considerable performance loss here.~~
After changing the approach of the isinf macro the performance loose should be fixed actually should be even faster because now it involves no datatype conversions
## Why It's Good For The Game
infinite values are bad

## Changelog
:cl:
refactor: refactors isinf macro
fix: fixes isinf for windows hosts
/:cl:

### CODE REVIEW

**Comment by ike709** [comment]
You should try making it a ternary using `world.system_type` to do either the windows test or the linux test.

One var access is presumably faster than 2 extra string comparisons.

**Comment by MNarath1** [comment]
> You should try making it a ternary using `world.system_type` to do either the windows test or the linux test.
> 
> One var access is presumably faster than 2 extra string comparisons.

i found a better solution that that and it works for linux and windows for some reason if you define 1.#INF and then try a compare check this works for linux and windows i am gonna commit that in a little bit

**Comment by MNarath1** [comment]
ok @ike709 this has been tested to work on windows and on linux thanks to kuberoot and unlike the previous solution it should be faster too because there is no conversion between datatypes needed unlike before

**Comment by MNarath1** [comment]
and there is no need for any extra tests only positive and negative infinity now why this works with windows and linux both even trough 1.#INF is a windows thing i have no idea honestly

File: code/__DEFINES/maths.dm [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 #define PI						3.1416
 #define INFINITY				1e31	//closer then enough
+#define inf_win_lin					1.#INF //only for isinf test
```
Comment (ike709) [inline code/__DEFINES/maths.dm]: Please use something more descriptive like `system_type_infinity`. That example in particular would match BYOND's existing `system_type` for specifying the OS of the host machine.

Comment (MNarath1) [reply code/__DEFINES/maths.dm]: ok

**Review by ike709 — DISMISSED** [review]

File: code/__DEFINES/maths.dm [hunk 2]
```diff
@@ -6,6 +6,7 @@
 
 #define PI						3.1416
 #define INFINITY				1e31	//closer then enough
+#define system_type_infinity					1.#INF //only for isinf check
```
Comment (PowerfulBacon) [inline code/__DEFINES/maths.dm]: Shouldnt we use full caps for this like other defines?

Comment (MNarath1) [reply code/__DEFINES/maths.dm]: You might be right at least for the sake of consistency.

**Comment by MNarath1** [comment]
no idea why this failed on the first attempt but seems like force pushing a commit so it reruns the checks did the trick.

**Review by ike709 — APPROVED** [review]
